### ISSUE TITLE
XML parser error for identityref type

### ISSUE THREAD

**Issue by ygorelik** [CESNET/libyang#703]
The libyang fails to parse perfectly valid XML file, which we get from Cisco IOS XR router:
```
<interfaces xmlns="http://openconfig.net/yang/interfaces">
  <interface>
    <name>TenGigE0/0/0/0</name>
    <config>
      <name>TenGigE0/0/0/0</name>
      <type xmlns:idx="urn:ietf:params:xml:ns:yang:iana-if-type">idx:ethernetCsmacd</type>
      <enabled>false</enabled>
    </config>
  </interface>
</interfaces>
```
The validation by yanglint (latest version 0.15.160) fails too:
```
yan@ubuntu-yan:~/ydk-workspace/temp/libyang/build$ ./yanglint
> add /home/yan/.ydk/172.27.150.154_830/openconfig-interfaces.yang
> data /home/yan/oc-int.xml
libyang[0]: Invalid value "idx:ethernetCsmacd" in "type" element. (path: /openconfig-interfaces:interfaces/interface[name='TenGigE0/0/0/0']/config/type)
Failed to parse data.
> quit
```
The file `oc-int.xml` contains above payload.
Will you be able to fix it?
What is the workaround?

**Comment by michalvasko** [comment]
Hi,
there is no problem problem in *libyang* but you have not loaded the module `iana-if-type` where the identity is defined.

Also, latest (released) version is 0.16.105.

Regards,
Michal

**Comment by ygorelik** [comment]
Hi Michal
The validation succeeds when iana-if-type module is loaded and all mandatory attributes are present in payload:
```
<interfaces xmlns="http://openconfig.net/yang/interfaces">
  <interface>
    <name>TenGigE0/0/0/0</name>
    <config>
      <name>TenGigE0/0/0/0</name>
      <type xmlns:idx="urn:ietf:params:xml:ns:yang:iana-if-type">idx:ethernetCsmacd</type>
      <enabled>false</enabled>
    </config>
    <state>
      <name>TenGigE0/0/0/0</name>
      <type xmlns:idx="urn:ietf:params:xml:ns:yang:iana-if-type">idx:ethernetCsmacd</type>
      <enabled>false</enabled>
      <admin-status>DOWN</admin-status>
      <oper-status>DOWN</oper-status>
    </state>
  </interface>
</interfaces>
```
Thank you,
Yan
